demo/src/app/home/home.page.ts

```diff
@@ -10,125 +17,196 @@ declare var FileTransferManager: any;
           } else {
             console.error("uploader caught an error: " + event.error);
           }
         } else if (event.state == 'UPLOADING') {
-          console.log("uploading: " + event.id + " progress: " + event.progress + "%");
-          correspondingMedia.updateStatus("uploading: " + event.progress + "%");
+          this.log("Uploading: " + event.id + " progress: " + event.progress + "%");
+          correspondingMedia.updateStatus("Uploading: " + event.progress + "%");
         }
-
+          
         if (event.eventId)
-          self.uploader.acknowledgeEvent(event.eventId);
+          this.uploader.acknowledgeEvent(event.eventId);
       });
     })
+
+    this.events.getUploadOptionsChange().subscribe(() =>  { this.loadUploadOptions() });
   }
 
-  private getMediaWithId(mediaId) {
-    for (var media of this.allMedia) {
-      if (media.id == mediaId) {
-        return media;
-      }
-    }
-    return null;
+  ngOnInit() {
+    this.loadUploadOptions();
   }
 
   cancelUpload(media: Media): void {
     this.uploader.removeUpload(media.id, res => {
-      console.log('removeUpload result: ', res);
       media.updateStatus("Aborted");
+      this.log("Upload: " + media.id + " aborted");
     }, err => alert('Error removing upload'));
   }
 
   openGallery(): void {
-    var self = this;
-
-    var options = {
-      width: 200,
-      quality: 25
-    };
-
-    self.imgPicker.getPictures({
+    this.imgPicker.getPictures({
       maximumImagesCount: 3
     }).then(file_uris => {
-      for (var i = 0; i < file_uris.length; i++) {
-        let path = this.win.Ionic.WebView.convertFileSrc(file_uris[i]);
-        var media = new Media(path, this._ngZone);
+      if(typeof file_uris == 'string') return;
+      file_uris.forEach(file_uri => {
+        const media = new Media(file_uri, this.webView.convertFileSrc(file_uri), this._ngZone);
         this.allMedia.push(media);
+        this.log("Upload: " + media.id + " added");
+      });
+      this.refreshRemainsMediaToUpload();
+    }, err => console.log('err: ' + err));
+  }
 
-        var options: any = {
-          serverUrl: "https://en7paaa03bwd.x.pipedream.net/",
-          filePath: file_uris[i],
-          fileKey: "file",
-          id: media.id,
-          notificationTitle: "Uploading image (Job 0)",
-          headers: {},
-          parameters: {
-            colors: 1,
-            faces: 1,
-            image_metadata: 1,
-            phash: 1,
-            signature: "924736486",
-            tags: "device_id_F13F74C5-4F03-B800-2F76D3C37B27",
-            timestamp: 1572858811,
-            type: "authenticated"
-          }
-        };
-        self.uploader.startUpload(options);
+  startUpload(media: Media) {
+    const options: any = {
+      serverUrl: this.uploadOptions.serverUrl,
+      filePath: media.uri,
+      fileKey: this.uploadOptions.fileKey,
+      id: media.id,
+      notificationTitle: "Uploading image (Job 0)",
```

Comment:
Pas de concept de Job sur la demo, 
```suggestion
      notificationTitle: "Uploading image",
```